### PR TITLE
Feature parse nmea function

### PR DESCRIPTION
**Pull request by itzandroidtab**

### CODE REVIEW

File: code/src/uart_nmea_c.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            wait_for_start,
+            get_message_id,
+            check_message_id,
+            get_sentence
+        };
+
+        get_location_state state = wait_for_start;
+        char uart_input = 0;
+
+        while (true) {
```
Comment (itzandroidtab) [inline code/src/uart_nmea_c.cpp]: this while loop should be removed

Comment (itzandroidtab) [reply code/src/uart_nmea_c.cpp]: We can probably keep it like this for now and fix it when the code reviewers want us to change this to master

File: code/headers/gga.hpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <hwlib.hpp>
+
+namespace r2d2::location {
+
+    struct gga_s {
```
Comment (Kaaserne) [inline code/headers/gga.hpp:13]: Maybe add a little bit of comment about what `gga_s` is, or its abbrevation.

File: code/headers/nmea_listener.hpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+         * @brief bool for storing if a string is avaialable
+         *
+         */
+        bool available;
+
+    public:
+        nmea_listener_c(usart::usart_connection_c &usart) : usart(usart) {
+        }
+
+        bool nmea_available() {
```
Comment (Kaaserne) [inline code/headers/nmea_listener.hpp]: This function, along with some others, may be `const`

File: code/main.cpp [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
+
+    r2d2::comm_c comm;
+
+    r2d2::usart::hardware_usart_c<r2d2::usart::usart0> usart(9600);
+
+    r2d2::location::uart_nmea_c nmea(usart);
+
+    r2d2::location::module_c module(comm, nmea);
+
+    while(true) {
```
Comment (Kaaserne) [inline code/main.cpp]: Style guide refers to `while (true)`. It is just a little nitpicky. One space between the `e` and the `(`

File: code/src/nmea_parser.cpp [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+namespace r2d2::location {
+    gga_s nmea_parser_c::parse_nmea(const uint8_t *gps_message, const size_t length) {
+        gga_s location;
+        GGA GGAindex = GGA::sentence;
+
+        for (size_t i = 0; i < length; i++){
+            size_t sep_offset = get_offset_separator(gps_message + i, length - i, ',');
+
+            switch (GGAindex){
+                case GGA::sentence: {
```
Comment (Kaaserne) [inline code/src/nmea_parser.cpp]: Shouldn't the `switch` statement be formatted as follows?
```cpp
switch (var) {
case GGA::sentence: {
    // stuff
    break;
}
case //bla
```

**Review by Kaaserne — CHANGES_REQUESTED** [review]

File: code/headers/nmea_listener.hpp (hunk 3 above)
Comment (itzandroidtab) [reply code/headers/nmea_listener.hpp]: made some of them constexpr

**Review by Kaaserne — APPROVED** [review]

**Pull request merged**